File: ariprog/ariprog.py
"""
ID: brkwok1
LANG: PYTHON3
TASK: ariprog
"""
import sys
sys.stderr.write('message')
from collections import defaultdict, deque

with open('ariprog.in', 'r') as fin:
    N = int(fin.readline().strip())
    M = int(fin.readline().strip())

    bisquares = set()
    for i in range(M + 1):
        for j in range(i, M + 1):
            bisquares.add(i*i + j*j)
    
    list_bs = sorted(list(bisquares))
    L = len(list_bs)
    res = []

    for i in range(L - 1, N - 2, -1):
        end = list_bs[i]
        diff = 1
        max_step = end // (N - 1)
        next_seq = list_bs[i - diff]
        curr_step = end - next_seq

        while curr_step <= max_step:
            is_inset = True

            for n in range(N - 2):
                next_seq -= curr_step

                if next_seq not in bisquares:
                    is_inset = False
                    break
                
            if is_inset:
                res.append((end - curr_step * (N - 1), curr_step))

            diff += 1
            next_seq = list_bs[i - diff]
            curr_step = end - next_seq

    res.sort(key= lambda x: (x[1], x[0]))

    with open('ariprog.out', 'w') as fout:
        if res:
            for x,y in res:
                fout.write(f'{x} {y}\n')
        else:
            fout.write('NONE\n')

File: ariprog/ariprog.cpp
/*
ID: brkwok1
LANG: C++11
TASK: ariprog
*/
#include <iostream>
#include <fstream>

#define MAX_BISQ 125001

using namespace std;

int bisquares[MAX_BISQ];
bool is_bisq[MAX_BISQ];

int main()
{
  ifstream fin("ariprog.in");
  ofstream fout("ariprog.out");

  int N, M, i, j, max_bs, max_seq, count = 0;

  fin >> N >> M;

  for (i = 0; i <= M; i++)
  {
    for (j = i; j <= M; j++)
    {
      int bs = i * i + j * j;
      bisquares[bs] = bs;
      is_bisq[bs] = true;
      max_bs = bs;
    }
  }

  max_bs += 1;

  max_seq = max_bs / (N - 1) + 2;

  for (int b = 1; b <= max_seq; b++)
  {
    for (int a = 0; a < max_bs; a++)
    {
      int x = a;

      for (i = 0; i < N; i++)
      {
        if (x >= max_bs || !is_bisq[x])
        {
          break;
        }

        x += b;
      }

      if (i == N)
      {
        fout << bisquares[a] << " " << b << "\n";
        count += 1;
      }
    }
  }

  if (count == 0)
    fout << "NONE\n";

  return 0;
}